movie_REST_API/movie_REST_API/Service/CoreDataService/CoreDataService.swift

```diff
@@ -0,0 +1,93 @@
+// CoreDataService.swift
+// Copyright © Natasha Ananas. All rights reserved.
+
+import CoreData
+import UIKit
+
+/// Сервис работы с CoreData
+final class CoreDataService: CoreDataServiceProtocol {
+    // MARK: - Constants
+
+    private enum Constants {
+        static let movieDataText = "MovieData"
+        static let emptyText = ""
+    }
+
+    // MARK: - Private Properties
+
+    private let context = (UIApplication.shared.delegate as? AppDelegate)?.persistentContainer.viewContext
+
+    // MARK: - Public Methods
+
+    func saveMovieCoreData(type: TypeMovie, movies: [Movie]) {
+        guard let context = context else { return }
+        context.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
+        guard let entity = NSEntityDescription.entity(forEntityName: Constants.movieDataText, in: context)
+        else { return }
+        saveMovie(entity: entity, context: context, movies: movies, type: type)
+        do {
+            try context.save()
+        } catch {}
+    }
+
+    func getMovieCoreData(type: TypeMovie) -> [Movie]? {
+        guard let context = context else { return nil }
+        let fetchRequest = NSFetchRequest<MovieData>(entityName: Constants.movieDataText)
+        let allMovies = try? context.fetch(fetchRequest)
+        guard let allMovies = allMovies else { return nil }
+        let movies = getMovie(allMovies: allMovies, type: type)
+        return movies
+    }
+
+    // MARK: - Private Methods
```

Comment:
```suggestion
    // MARK: - Private Methods
```
перемести марку ниже на 2 метода